apps/dashboard/src/app/api/testnet-faucet/can-claim/route.ts

```diff
@@ -0,0 +1,54 @@
+import { getIpAddress } from "lib/ip";
+import { cacheTtl } from "lib/redis";
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+const THIRDWEB_ENGINE_URL = process.env.THIRDWEB_ENGINE_URL;
+const NEXT_PUBLIC_THIRDWEB_ENGINE_FAUCET_WALLET =
+  process.env.NEXT_PUBLIC_THIRDWEB_ENGINE_FAUCET_WALLET;
+const THIRDWEB_ACCESS_TOKEN = process.env.THIRDWEB_ACCESS_TOKEN;
+
+export type CanClaimResponseType =
+  | {
+      canClaim: boolean;
+      ttlSeconds: number;
+      type: "throttle";
+    }
+  | {
+      canClaim: false;
+      type: "unsupported-chain";
+    };
+
+// Note: This handler cannot use "edge" runtime because of Redis usage.
+export const GET = async (req: NextRequest) => {
+  const searchParams = req.nextUrl.searchParams;
+  const chainId = searchParams.get("chainId");
+
+  if (Number.isNaN(chainId)) {
```

Comment:
uhh what does `Number.isNaN` do with strings? because chainId will be a string here, no?

Reply:
searchParams.get returns a `string | null` and not `number`

Reply:
hmm, actually - this check here is not enough.

Reply:
Updated